src/App.tsx

```diff
@@ -1,61 +1,149 @@
+import { useState, FormEvent } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './types/PreparedTodo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getNewTodoId = (todoList: PreparedTodo[]) => {
+  const usedIds = todoList.map(todo => todo.id).sort((a, b) => a - b);
+  const freeId = usedIds.findIndex((usId, index) => usId !== index + 1);
+  const newId = freeId !== -1 ? freeId + 1 : usedIds.length + 1;
+
+  return newId;
+};
 
 export const App = () => {
+  const initialTodoList: PreparedTodo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId) || null,
+  }));
+
+  const [titleInput, setTitleInput] = useState('');
+  const [userIdSelect, setUserIdSelect] = useState(0);
+  const [preparedTodoList, setPreparedTodoList] = useState(initialTodoList);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleEnter = (value: string) => {
+    setTitleInput(value);
+    if (titleError === true) {
+      setTitleError(false);
+    }
+  };
+
+  const handleUserSelect = (value: number) => {
+    setUserIdSelect(value);
+    if (userError === true) {
+      setUserError(false);
+    }
+  };
+
+  const validateForm = () => {
+    const noTitle = titleInput === '';
+    const noUser = userIdSelect === 0;
+
+    if (noTitle || noUser) {
+      if (noTitle) {
+        setTitleError(true);
+      }
+
+      if (noUser) {
+        setUserError(true);
+      }
+
+      return false;
+    }
+
+    return true;
+  };
+
+  const clearForm = () => {
+    setTitleInput('');
+    setUserIdSelect(0);
+  };
+
+  const handleFormSubmit = (event: FormEvent) => {
```

Comment:
you can make it shortest
```suggestion
  const handleSubmit = (event: FormEvent) => {
```